xdsl/ir.py

```diff
@@ -855,22 +855,36 @@ def irdl_definition(cls) -> OpDef | None:
 OperationInvT = TypeVar('OperationInvT', bound=Operation)
 
 
-@dataclass()
+@dataclass(init=False)
 class Block(IRNode):
     """A sequence of operations"""
 
-    declared_at: Span | None = None
+    declared_at: Span | None
 
-    _args: tuple[BlockArgument, ...] = field(default_factory=lambda: (),
-                                             init=False)
+    _args: tuple[BlockArgument, ...]
     """The basic block arguments."""
 
-    ops: list[Operation] = field(default_factory=list, init=False)
+    ops: list[Operation]
     """Ordered operations contained in the block."""
 
-    parent: Region | None = field(default=None, init=False, repr=False)
+    parent: Region | None
     """Parent region containing the block."""
 
+    def __init__(self,
+                 ops: Iterable[Operation] = (),
+                 arg_types: tuple[Attribute, ...] = (),
```

Comment:
Can we remain consistent with `Region` and make `parent` and `declared_at` kw_only?